src/implementation/abstractarray.jl

```diff
@@ -82,39 +106,116 @@ end
 # Implementation based on Base + LinearAlgebra
 #-------------------------------------------------------------------------------------------
 # Note that this is mostly for convenience + checking, and not for performance
+function tensoradd!(C::AbstractArray,
+                    A::AbstractArray, pA::Index2Tuple, conjA::Bool,
+                    α::Number, β::Number, ::BaseView)
+    argcheck_tensoradd(C, A, pA)
+    dimcheck_tensoradd(C, A, pA)
 
+    # can we assume that C is mutable?
+    # is there more functionality in base that we can use?
+    if conjA
+        if iszero(β)
+            C .= α .* conj.(PermutedDimsArray(A, linearize(pA)))
+        else
+            C .= β .* C .+ α .* conj.(PermutedDimsArray(A, linearize(pA)))
+        end
+    else
+        if iszero(β)
+            C .= α .* PermutedDimsArray(A, linearize(pA))
+        else
+            C .= β .* C .+ α .* PermutedDimsArray(A, linearize(pA))
+        end
+    end
+    return C
+end
 function tensoradd!(C::AbstractArray,
                     A::AbstractArray, pA::Index2Tuple, conjA::Bool,
-                    α::Number, β::Number)
+                    α::Number, β::Number, ::BaseCopy)
     argcheck_tensoradd(C, A, pA)
     dimcheck_tensoradd(C, A, pA)
 
     # can we assume that C is mutable?
     # is there more functionality in base that we can use?
```

Comment:
I think there is an in-place `permutedims!`, such that we could use the allocation interface to also hijack into allocating these temporary arrays

Reply:
That might be possible indeed. The question is whether it is worth it. This will probably only be used for types which are very different from strided arrays, e.g. sparse arrays.

Reply:
I think that's probably a very valid point. It's probably fair to assume that we cannot guarantee optimal performance without extra information about the specific type anyways, and the system does allow to easily implement custom backends if necessary. The base backend should serve mostly as a catch-all implementation that ensures that it works for most types.